provider/cluster/inventory.go

```diff
@@ -450,6 +448,10 @@ loop:
 				is.log.Info("removing reservation", "order", res.OrderID())
 
 				reservations = append(reservations[:idx], reservations[idx+1:]...)
+				// reclaim availableExternalPorts if unreserving allocated resources
+				if res.allocated {
+					is.availableExternalPorts += reservationCountEndpoints(res)
```

Comment:
looks reasonable, but I defer to @hydrogen18 on this

Reply:
Makes sense to me, I just missed the case. When the reservation is has `allocated = true` the code does `is.availableExternalPorts -= externalPortCount` so this restores the previous value when the reservation is released _if it has been allocated_.

It looks like @arijitAD added a test for this so :+1: :100: